### PR TITLE
add login_method to user from token, add to tests, bump minor version

### PR DESCRIPTION
**Pull request by mrmauer**
[ticket](https://propelauth.atlassian.net/browse/TIC-795)
[BE change](https://github.com/PropelAuth/auth/pull/576/files#diff-cc3fcb0282c1c7ba1da6a811ee1a6338e42c17fc21e17cb4c1bf05c30ec7b894)

Tests
added to unit tests which all pass

### CODE REVIEW

**Review by itailevi98 — DISMISSED** [review]
lgtm!

**Review by itailevi98 — COMMENTED** [review]
Undoing my approval, as I have now committed to the PR

**Comment by itailevi98** [comment]
@mrmauer @andrew-propelauth I've added types for the `login_method`. Since we want to enforce type safety, we do this in the form of classes, where each type is a class. See the `test_user_conversion` tests for examples.

File: propelauth_py/types/login_method.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
+from enum import Enum
+
+
+class SocialLoginProvider(Enum):
+    GOOGLE = "Google"
+    GITHUB = "GitHub"
+    SLACK = "Slack"
```
Comment (andrew-propelauth) [inline propelauth_py/types/login_method.py:7]: Missing `Microsoft`

**Review by andrew-propelauth — APPROVED** [review]
Lgtm, just missing microsoft, I'm gonna add it in

**Pull request merged**